src/commands/operator/index.ts

```diff
@@ -581,47 +474,58 @@ export default class Operator extends Command {
-      const operatorPrompt: any = await inquirer.prompt([
-        {
-          name: 'shouldContinue',
-          message: `A job is available for execution, would you like to operate?\n`,
-          type: 'confirm',
-          default: true,
-        },
-      ])
-      operate = operatorPrompt.shouldContinue
-    }
+    // quickly check that job is still valid
+    await this.checkJobStatus(jobHash)
+    if (jobHash in this.operatorJobs) {
+      const job: OperatorJob = this.operatorJobs[jobHash]
+      const network: string = job.network
+      let operate = this.operatorMode === OperatorMode.auto
+      if (this.operatorMode === OperatorMode.manual) {
+        const operatorPrompt: any = await inquirer.prompt([
+          {
+            name: 'shouldContinue',
+            message: `A job is available for execution, would you like to operate?\n`,
+            type: 'confirm',
+            default: true,
+          },
+        ])
+        operate = operatorPrompt.shouldContinue
+      }
 
-    if (operate) {
-      const gasPricing: GasPricing = this.networkMonitor.gasPrices[network]
-      const gasPrice: BigNumber = gasPricing.isEip1559 ? gasPricing.maxFeePerGas! : gasPricing.gasPrice!
-
-      const receipt: TransactionReceipt | null = await this.networkMonitor.executeTransaction({
-        network,
-        tags,
-        contract: this.networkMonitor.operatorContract,
-        methodName: 'executeJob',
-        args: [job.payload],
-        gasPrice: gasPrice,
-        gasLimit: job.gasLimit.mul(BigNumber.from('2')),
-      })
-      return receipt !== null
+      if (operate) {
+        const gasPricing: GasPricing = this.networkMonitor.gasPrices[network]
+        const gasPrice: BigNumber = gasPricing.isEip1559 ? gasPricing.maxFeePerGas! : gasPricing.gasPrice!
+
+        const receipt: TransactionReceipt | null = await this.networkMonitor.executeTransaction({
+          network,
+          tags,
+          contract: this.networkMonitor.operatorContract,
+          methodName: 'executeJob',
+          args: [job.payload],
+          gasPrice: gasPrice,
+          gasLimit: BigNumber.from(job.gasLimit).mul(BigNumber.from('2')),
+          canFail: false,
+          waitForReceipt: true,
+          interval: 1000,
+        })
+        if (receipt !== null && receipt.status === 1) {
+          delete this.operatorJobs[jobHash]
+        }
+
+        return receipt !== null
+      }
+
+      this.networkMonitor.structuredLog(network, 'Available job will not be executed', tags)
+      return false
     }
 
-    this.networkMonitor.structuredLog(network, 'Available job will not be executed', tags)
-    return false
+    return true
   }
 
   /**
    * Execute the lz message payload on the destination network
    */
   async executeLzPayload(network: string, jobHash: string, args: any[], tags: (string | number)[]): Promise<void> {
-    this.networkMonitor.walletNonces[network] = await this.networkMonitor.wallets[network].getTransactionCount()
```

Comment:
no longer needed?

Reply:
Nope, executeTransaction function contains wallet nonce logic within itself, this is just dead code at this point.